app/controllers/posts_controller.rb

```diff
@@ -1,6 +1,5 @@
 class PostsController < ApplicationController
-  before_action :set_post, only: %i[ show edit update destroy ]
-  http_basic_authenticate_with :name => "dhh", :password => "secret", :except => [:index, :show]
+  before_action :authenticate_user!, only: %i[new create]
```

Comment:
Can't find the authenticate_user! method anywhere in the project.
You should have a method similar to the one below in the `applicationController` helper. And then use it just the way you have done in the `PostsController`.
* [ ] it uses `before_action` to restrict access to the `:new` and `:create` methods (uses ApplicationController helper method) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
```
def signed_in_only!
 #current_user in ApplicationController must be defined as helper method to make it work

   redirect_to  new_user_session_path unless current_user

  # alternative solution:
    if signed_in? is defined in ApplicationController 
    redirect_to new_user_session_path unless signed_in?
end
```